rpc/namespaces/ethereum/eth/filters/api.go

```diff
@@ -350,16 +350,22 @@ func (api *PublicFilterAPI) NewHeads(ctx context.Context) (*rpc.Subscription, er
 					return
 				}
 
-				data, ok := ev.Data.(tmtypes.EventDataNewBlockHeader)
+				data, ok := ev.Data.(tmtypes.EventDataNewBlockEvents)
 				if !ok {
-					api.logger.Debug("event data type mismatch", "type", fmt.Sprintf("%T", ev.Data))
+					api.logger.Debug("EventDataNewBlockEvents type mismatch", "type", fmt.Sprintf("%T", ev.Data))
 					continue
 				}
```

Comment:
question; does it work well?

Reply:
i am not sure that `ev.Data.(tmtypes.EventDataNewBlockEvents)` can be succeeded.

Reply:
I think my understanding of that code was lacking. After a little more research, I've modified it in an appropriate way, and I think it should work fine. As an aside, I don't think that API function will ever be called externally. The NewHeads function is not defined in the FilterAPI interface.

Reply:
@zsystm 
I fixed pubSubAPI.subscribeNewHeads